src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateImplicitNamespaceImports.targets

```diff
@@ -0,0 +1,66 @@
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--
+    ============================================================
+                                     GenerateImplicitNamespaceImports
+
+    Generates implicit namespace imports source to intermediate directory for C# projects
+    ============================================================
+    -->
+  <PropertyGroup>
+    <GeneratedImplicitNamespaceImportFile Condition="'$(GeneratedImplicitNamespaceImportFile)' ==''">$(IntermediateOutputPath)$(MSBuildProjectName).ImplicitNamespaceImports$(DefaultLanguageSourceExtension)</GeneratedImplicitNamespaceImportFile>
+    <DisableImplicitNamespaceImports_DotNet Condition="'$(DisableImplicitNamespaceImports_DotNet)' == '' and '$(DisableImplicitNamespaceImports)' != 'true'">false</DisableImplicitNamespaceImports_DotNet>
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(DisableImplicitNamespaceImports_DotNet)' != 'true'
+                      and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'
+                      and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '6.0'))">
+    <Import Include="System" />
+    <Import Include="System.Collections.Generic" />
+    <Import Include="System.IO" />
+    <Import Include="System.Linq" />
+    <Import Include="System.Net.Http" />
+    <Import Include="System.Threading.Tasks" />
+  </ItemGroup>
+
+  <!--
+    Note that this must run before every invocation of CoreCompile to ensure that all compiler
+    runs see the generated assembly info. There is at least one scenario involving Xaml
+    where CoreCompile is invoked without other potential hooks such as Compile or CoreBuild,
+    etc., so we hook directly on to CoreCompile. Furthermore, we  must run *after*
+    PrepareForBuild to ensure that the intermediate directory has been created.
+
+    Targets that generate Compile items are also expected to run before
+    BeforeCompile targets (common targets convention).
+   -->
+  <Target Name="GenerateImplicitNamespaceImports"
+          BeforeTargets="BeforeCompile;CoreCompile"
+          DependsOnTargets="PrepareForBuild;CoreGenerateImplicitNamespaceImports"
+          Condition="'$(DisableImplicitNamespaceImports)' != 'true'" />
+
+  <Target Name="CoreGenerateImplicitNamespaceImports"
+          Outputs="$(GeneratedImplicitNamespaceImportFile)">
+    <ItemGroup>
+      <_ImportFileLine Include="// %3Cautogenerated />"/>
+      <_ImportFileLine Include="global using global::%(Import.Identity)%3B"/>
```

Comment:
How would we handle duplicates? I'd go with a distinct option. Otherwise we need to have a meaningful error message.

Reply:
I can add a warning for duplicated names and use a distinct list.